lib/views/home_content/feed/post_card/votes_widget.dart

```diff
@@ -1,43 +1,59 @@
 import "package:flutter/material.dart";
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:proxima/models/database/post/post_id_firestore.dart";
+import "package:proxima/models/database/vote/upvote_state.dart";
+import "package:proxima/viewmodels/upvote_view_model.dart";
 
 /// This widget is used to display the votes of a post.
 /// It contains two buttons to upvote and downvote the post and the number of votes.
-class VotesWidget extends StatelessWidget {
-  final int votes;
+class VotesWidget extends HookConsumerWidget {
+  final PostIdFirestore postId;
 
   const VotesWidget({
     super.key,
-    required this.votes,
+    required this.postId,
   });
 
   @override
-  Widget build(BuildContext context) {
+  Widget build(BuildContext context, WidgetRef ref) {
+    final asyncPostVote = ref.watch(postVoteProvider((postId: postId)));
+    final votes = asyncPostVote.valueOrNull?.votes ?? 0;
+    final upvoteState =
+        asyncPostVote.valueOrNull?.upvoteState ?? UpvoteState.none;
+
+    final upvoteNotifier =
+        ref.read(postVoteProvider((postId: postId)).notifier);
+
     final upvote = IconButton(
       //Used to reduce the padding and InkWell created by the IconButton widget
       //https://stackoverflow.com/questions/50381157/how-do-i-remove-flutter-iconbutton-big-padding
       padding: EdgeInsets.zero,
       constraints: const BoxConstraints(),
-      icon: const Icon(Icons.expand_less, size: 30),
+      icon: Icon(
+        Icons.expand_less,
+        size: 30,
+        color: upvoteState == UpvoteState.upvoted ? Colors.blue : null,
+      ),
       style: const ButtonStyle(
         tapTargetSize: MaterialTapTargetSize.shrinkWrap,
       ),
-      onPressed: () {
-        //TODO: Implement the logic to upvote the post
-      },
+      onPressed: () => upvoteNotifier.triggerUpVote(),
     );
 
     final downvote = IconButton(
       //Used to reduce the padding and InkWell created by the IconButton widget
```

Comment:
Avoid duplicating comments as it makes them harder to maintain.

Reply:
You are right, I remove the duplication in 63fecb17c93cd8cb541334a9ecd53d0d851ba7bc.